slam-libraries/viam-orb-slam3/orbslam_server_v1.cc

```diff
@@ -778,13 +779,24 @@ double ReadTimeFromFilename(string filename) {
 
     // Now we read from buffer using get_time manipulator
     // and formatting the input appropriately.
-    ss >> std::get_time(&dt, "%Y-%m-%dT%H_%M_%SZ");
-    double sub_sec =
-        (double)std::stof(filename.substr(filename.find(".")), &sz);
+    ss >> std::get_time(&dt, time_format.c_str());
     time_t thisTime = std::mktime(&dt);
+    auto sub_sec_index = filename.find(".");
+    if ((sub_sec_index != string::npos)) {
+        double sub_sec = (double)std::stof(filename.substr(sub_sec_index), &sz);
+        double myTime = (double)thisTime + sub_sec;
+        return myTime;
+    } else {
+        return (double)thisTime;
+    }
+}
```

Comment:
Would it be possible to put this also into a `ReadTimeFromFilename` function, together with the `string time_format` definition above?

Reply:
these are currently inside the ReadTimeFromFilename function

Reply:
as for `string time_format` I do not want to move that into the function, since multiple functions depend on it